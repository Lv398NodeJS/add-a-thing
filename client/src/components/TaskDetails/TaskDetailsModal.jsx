import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import TaskDetails from './TaskDetails';
import SubtaskListContainer from '../Subtask/SubtaskListContainer';
import DeleteTask from './DeleteTask';

export default class TaskDetailsModal extends React.Component {
  constructor() {
    super();
    this.state = {
      showDelete: false,
    };
  }

  closeTaskDetails = () => {
    const { onClose: close } = this.props;
    close();
  };

  closeDeleteModal = () => {
    const { showDelete } = this.state;
    this.setState({
      showDelete: !showDelete,
    });
  };

  render() {
    const { show: modalShow } = this.props;
    const { showDelete } = this.state;
    return (
      <Modal
        show={modalShow}
        onHide={this.closeTaskDetails}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Body>
          <TaskDetails
            closeTaskDetails={this.closeTaskDetails}
          />
          <SubtaskListContainer />
        </Modal.Body>
        <Button
          variant="outline-danger"
          onClick={() => this.setState({ showDelete: !showDelete })}
          className="delete-task-details-button float-right"
        >
          {'Delete'}
        </Button>
        <DeleteTask
          showDelete={showDelete}
          closeTaskDetails={this.closeTaskDetails}
          closeDeleteModal={this.closeDeleteModal}
        />
      </Modal>
    );
  }
}
