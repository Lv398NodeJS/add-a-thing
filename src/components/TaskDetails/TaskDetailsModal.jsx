import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import TaskDetails from './TaskDetails';
import SubTaskList from '../SubTaskList/SubTaskList';
import DeleteTask from './DeleteTask';

export default class TaskDetailsModal extends React.Component {
  constructor() {
    super();
    this.state = {
      showDelete: false,
    };
  }

  closeTaskDetails = () => {
    const { onClose: close } = this.props;
    close();
  };

  closeDeleteModal = () => {
    const { showDelete } = this.state;
    this.setState({
      showDelete: !showDelete,
    });
  };

  render() {
    const { taskRef, show: modalShow } = this.props;
    const { showDelete } = this.state;
    return (
      <Modal
        show={modalShow}
        onHide={this.closeTaskDetails}
        aria-labelledby="contained-modal-title-center"
        centered
      >
        <Modal.Body>
          <TaskDetails
            taskRef={taskRef}
            closeTaskDetails={this.closeTaskDetails}
          />
          <SubTaskList taskRef={taskRef} />
        </Modal.Body>
        <Button
          variant="outline-danger"
          onClick={() => this.setState({ showDelete: !showDelete })}
          className="delete-task-details-button float-right"
        >
          {'Delete'}
        </Button>
        <DeleteTask
          taskRef={taskRef}
          showDelete={showDelete}
          closeTaskDetails={this.closeTaskDetails}
          closeDeleteModal={this.closeDeleteModal}
        />
      </Modal>
    );
  }
}
