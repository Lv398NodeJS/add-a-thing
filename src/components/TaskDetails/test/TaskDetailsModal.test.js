import React from 'react';
import { shallow } from 'enzyme';
import TaskDetailsModal from '../TaskDetailsModal';
// import DeleteTask from '../DeleteTask';

describe('TaskDetailsModal', () => {
  const component = shallow(<TaskDetailsModal />);
  it('should render Modal window correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render DeleteTask child component', () => {
    expect(component.find('DeleteTask').length).toBe(1);
  });

  it('should render TaskDetails child component', () => {
    expect(component.find('TaskDetails').length).toBe(1);
  });

  it('should render SubTaskList child component', () => {
    expect(component.find('SubTaskList').length).toBe(1);
  });
});
