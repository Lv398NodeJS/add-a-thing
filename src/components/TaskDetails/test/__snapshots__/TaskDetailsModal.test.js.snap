// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetailsModal should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskDetailsModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "aria-labelledby": "contained-modal-title-center",
      "centered": true,
      "children": Array [
        <ForwardRef
          className="MainModalTD"
          closeButton={false}
          closeLabel="Close"
        >
          Task details
          <ForwardRef
            active={false}
            aria-label="Close"
            className="close"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <TaskDetails />
          <ForwardRef
            as="div"
            fluid={false}
          >
            <SubTaskList />
          </ForwardRef>
        </ForwardRef>,
      ],
      "onHide": [Function],
      "show": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Task details",
            <ForwardRef
              active={false}
              aria-label="Close"
              className="close"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </ForwardRef>,
          ],
          "className": "MainModalTD",
          "closeButton": false,
          "closeLabel": "Close",
        },
        "ref": null,
        "rendered": Array [
          "Task details",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "aria-label": "Close",
              "children": <span
                aria-hidden="true"
              >
                ×
              </span>,
              "className": "close",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "variant": "primary",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "children": "×",
              },
              "ref": null,
              "rendered": "×",
              "type": "span",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <TaskDetails />,
            <ForwardRef
              as="div"
              fluid={false}
            >
              <SubTaskList />
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "taskRef": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <SubTaskList />,
              "fluid": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "taskRef": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
                "fluid": false,
              },
              "displayName": "Container",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "BACKDROP_TRANSITION_DURATION": 150,
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalBody",
        "render": [Function],
      },
      "Dialog": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "ModalDialog",
        "render": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalFooter",
        "render": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "closeButton": false,
          "closeLabel": "Close",
        },
        "displayName": "ModalHeader",
        "render": [Function],
      },
      "TRANSITION_DURATION": 300,
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalTitle",
        "render": [Function],
      },
      "defaultProps": undefined,
      "displayName": "Bootstrap(Modal)",
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-labelledby": "contained-modal-title-center",
        "centered": true,
        "children": Array [
          <ForwardRef
            className="MainModalTD"
            closeButton={false}
            closeLabel="Close"
          >
            Task details
            <ForwardRef
              active={false}
              aria-label="Close"
              className="close"
              disabled={false}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <TaskDetails />
            <ForwardRef
              as="div"
              fluid={false}
            >
              <SubTaskList />
            </ForwardRef>
          </ForwardRef>,
        ],
        "onHide": [Function],
        "show": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Task details",
              <ForwardRef
                active={false}
                aria-label="Close"
                className="close"
                disabled={false}
                onClick={[Function]}
                type="button"
                variant="primary"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </ForwardRef>,
            ],
            "className": "MainModalTD",
            "closeButton": false,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Array [
            "Task details",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "aria-label": "Close",
                "children": <span
                  aria-hidden="true"
                >
                  ×
                </span>,
                "className": "close",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
                "variant": "primary",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "children": "×",
                },
                "ref": null,
                "rendered": "×",
                "type": "span",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "closeButton": false,
              "closeLabel": "Close",
            },
            "displayName": "ModalHeader",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <TaskDetails />,
              <ForwardRef
                as="div"
                fluid={false}
              >
                <SubTaskList />
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "taskRef": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": <SubTaskList />,
                "fluid": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "taskRef": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                  "fluid": false,
                },
                "displayName": "Container",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalBody",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "BACKDROP_TRANSITION_DURATION": 150,
        "Body": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
        "Dialog": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        },
        "Footer": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalFooter",
          "render": [Function],
        },
        "Header": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
        "TRANSITION_DURATION": 300,
        "Title": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalTitle",
          "render": [Function],
        },
        "defaultProps": undefined,
        "displayName": "Bootstrap(Modal)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
